main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt
+package plugins
+
+import Keys._
+import sbt.librarymanagement.syntax._
+import sbt.librarymanagement.CrossVersion
+import Project.inConfig
+
+object SemanticdbPlugin extends AutoPlugin {
+  override def requires = JvmPlugin
+  override def trigger = allRequirements
+
+  override lazy val globalSettings = Seq(
+    semanticdbEnabled := false,
+    semanticdbIncludeInJar := false,
+    semanticdbOptions := List("-Yrangepos"),
+    semanticdbVersion := "4.0.0",
+    semanticdbCompilerPlugin := {
+      val v = semanticdbVersion.value
+      ("org.scalameta" % "semanticdb-scalac" % v).cross(CrossVersion.full)
+    }
+  )
+
+  override lazy val projectSettings = Seq(
+    allDependencies ++= {
```

Comment:
I've never seen `allDependencies`, the documentation says "Inter-project and library dependencies". How does this workaround users who do `libraryDependencies :=`?

Reply:
What about builds that do `scalacOptions :=`? I think that is more common than hardcoding library dependencies.

Reply:
`allDependencies` is an sbt-internal setting that we use to append other things onto `libraryDependencies`, and that's what we actually pass to the library management engine.
https://github.com/sbt/sbt/blob/v1.2.4/main/src/main/scala/sbt/Defaults.scala#L2268-L2279
So even if the build user sets the `libraryDependencies` to `Nil` it doesn't affect `allDependencies`. Normally we name keys `fullSomething`.

I can probably define something similar for `scalacOptions`.